docker/docker_image_dest.go

```diff
@@ -31,6 +32,7 @@ type dockerImageDestination struct {
 	c   *dockerClient
 	// State
 	manifestDigest digest.Digest // or "" if not yet known.
+	mutex          sync.Mutex
```

Comment:
Please always explicitly document what data is protected by each individual synchronization mechanism.